tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -653,6 +653,23 @@ mod test {
         });
     }
 
+    #[test]
+    fn json_span_event() {
+        // Check span events serialize correctly.
+        // Discussion: https://github.com/tokio-rs/tracing/issues/829#issuecomment-661984255
+        //
+        let expected = r#"{"timestamp":"fake time","level":"INFO","fields":{"message":"enter"},"target":"tracing_subscriber::fmt::format::json::test"}"#;
+        let collector = collector()
+            .flatten_event(false)
+            .with_current_span(false)
+            .with_span_list(false)
+            .with_span_events(FmtSpan::ENTER);
+
+        test_json(expected, collector, || {
+            tracing::info_span!("valid_json").in_scope(|| {});
+        });
+    }
+
```

Comment:
take it or leave it, but since the issue here is that we were generating invalid JSON, it _might_ be worth rewriting this test in the same style as the [`record_works` test](https://github.com/tokio-rs/tracing/blob/0a49b8d0d11e54eff0a55a62bcba39c2ef87ada1/tracing-subscriber/src/fmt/format/json.rs#L611-L654)? this test actually attempts to parse the output as JSON and then makes assertions about its structure. this could be worth doing, since it would also implicitly assert that the output definitely does parse as JSON.

of course, since the hard-coded JSON string _is_ valid JSON as well, this test is also fine as-is.